### PR TITLE
[# 167242781] TP can be accessed by members via md-search url

### PR DESCRIPTION
**Pull request by kartik-gr**
TP can be accessed by members via md-search url

This is the first phase of the update - modifying the SSO gem to allow authorization by role. I will also be updating TP to use the new module (as well as a few other applications that use DJ for which I'm breaking the AuthorizationConstraint interface).

[Tracker #167242781](https://www.pivotaltracker.com/story/show/167242781)

### CODE REVIEW

File: lib/grnds/sso/authorization.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Grnds
+  module Sso
+    module Authorization
+      extend ActiveSupport::Concern
+
+      included do
+        helper_method :authorized_for?
+      end
+
+      def authorized_for?(request, roles)
```
Comment (gauravs) [inline lib/grnds/sso/authorization.rb]: The parameters seem to be reversed from the expected ones: https://github.com/ConsultingMD/tp/pull/1247/files#diff-55c5b7aecfb519d0e4880eaf2788eb6eR35

File: lib/grnds/sso/authorization_constraint.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 module Grnds::Sso
   class AuthorizationConstraint
+    attr_accessor :roles
```
Comment (gauravs) [inline lib/grnds/sso/authorization_constraint.rb]: Nit, does this need to be a `attr_accessor` ?

File: lib/grnds/sso/authorization.rb (hunk 1 above)
Comment (richddr) [reply lib/grnds/sso/authorization.rb]: good catch

Comment (richddr) [reply lib/grnds/sso/authorization.rb]: but I guess that can be fixed either here or in the client PRs

File: lib/grnds/sso/authorization_constraint.rb (hunk 2 above)
Comment (kartik-gr) [reply lib/grnds/sso/authorization_constraint.rb]: Probably not, I'll change it.

File: lib/grnds/sso/authorization.rb (hunk 1 above)
Comment (kartik-gr) [reply lib/grnds/sso/authorization.rb]: Yeah, I changed it here but didn't in the TP PR. So it's as it should be here.

File: lib/grnds/sso/authorization_constraint.rb [hunk 3]
```diff
@@ -17,7 +16,7 @@ def authenticated?(request)
     def authorized?(request)
       session = request.session
       return true if session['primary_role'] == 'admin'
-      return session['primary_role'] == (self.role)
+      return roles.include?(session['primary_role'])
```
Comment (gauravs) [inline lib/grnds/sso/authorization_constraint.rb]: Nit, avoids repeated statements this way:

```ruby
def authorized?(request)
  primary_role = request.session['primary_role']
  primary_role == 'admin' || @roles.include?(primary_role)
end
```

**Comment by soonahn** [comment]
@cobbr2 @richddr :wave:

File: lib/grnds/sso/authorization_constraint.rb (hunk 3 above)
Comment (kartik-gr) [reply lib/grnds/sso/authorization_constraint.rb]: I changed the client code to explicitly pass in the role for this very reason. The first line is unnecessary, I'll remove it.

**Comment by kartik-gr** [comment]
@gauravs no tests to be found anywhere around these parts. :(

Probably a worthwhile change, but I'm going to punt on it since I'd like to get this change out soon-ish.

**Comment by gauravs** [comment]
@kartik-gr sounds good. I'll try to take some time this week to setup some rspec scaffolding in this project

**Pull request merged**